tests/test_basic_ops_list.py

```diff
@@ -87,13 +88,33 @@ def test_items_with_interpolation() -> None:
     assert c == ["foo", "foo"]
 
 
-def test_list_pop() -> None:
-    c = OmegaConf.create([1, 2, 3, 4])
-    assert c.pop(0) == 1
-    assert c.pop() == 4
-    assert c == [2, 3]
-    with pytest.raises(IndexError):
-        c.pop(100)
+@pytest.mark.parametrize(
+    ["cfg", "key", "expected_out", "expected_cfg"],
+    [
+        pytest.param([1, 2, 3], 0, 1, [2, 3]),
+        pytest.param([1, 2, 3], None, 3, [1, 2]),
+        pytest.param([1, 2, 3], 100, IndexError, ...),
+        pytest.param(["???", 2, 3], 0, None, [2, 3]),
+        pytest.param(["${4}", 2, 3], 0, InterpolationKeyError, ...),
+        pytest.param(["${1}", "???", 3], 0, InterpolationToMissingValueError, ...),
+    ],
+)
+def test_list_pop(
+    cfg: List[Any], key: Optional[int], expected_out: Any, expected_cfg: Any
+) -> None:
+    c = OmegaConf.create(cfg)
+    pop_args = [] if key is None else [key]
+    if isinstance(expected_out, type):
+        with pytest.raises(expected_out):
+            c.pop(*pop_args)
+    else:
+        val = c.pop(*pop_args)
+        assert val == expected_out
+
+    if expected_cfg is ...:  # shortcut to mean it is unchanged
+        expected_cfg = cfg
+    assert c == expected_cfg
+
```

Comment:
can you split to errors testing to a different test?

Reply:
Done in 2dd491e